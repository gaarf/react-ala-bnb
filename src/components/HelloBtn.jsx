import React from 'react';
import PropTypes from 'prop-types';
import { withStyles, css, withStylesPropTypes } from '../helpers/withStyles';

function HelloBtn({ styles, children, onClick }) {
  return (
    <button type="submit" onClick={onClick} {...css(styles.btn)}>
      {children}
    </button>
  );
}

HelloBtn.propTypes = {
  ...withStylesPropTypes,
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default withStyles(() => ({
  btn: {
    fontSize: '3em',
    margin: 1,
    padding: `.2em 1em`,
    ':hover': {
      backgroundColor: 'orange',
    },
  },
}))(HelloBtn);
