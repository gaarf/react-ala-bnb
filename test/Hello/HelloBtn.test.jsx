import React from 'react';
import { shallow } from 'enzyme';
import HelloBtn from '../../src/components/HelloBtn';

describe('HelloBtn', () => {
  let ctnr;

  beforeEach(() => {
    ctnr = shallow(<HelloBtn onClick={jest.fn()}>hi</HelloBtn>);
  });

  it('looks alright', () => {
    expect(ctnr).toMatchSnapshot();
  });
});
