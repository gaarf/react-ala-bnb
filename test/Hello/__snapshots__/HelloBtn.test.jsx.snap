// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloBtn looks alright 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withStyles(HelloBtn)
    onClick={[MockFunction]}
  >
    hi
  </withStyles(HelloBtn)>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": "hi",
      "css": [Function],
      "onClick": [MockFunction],
      "styles": Object {
        "btn": Object {
          "_definition": Object {
            ":hover": Object {
              "backgroundColor": "orange",
            },
            "fontSize": "3em",
            "margin": 1,
            "padding": ".2em 1em",
          },
          "_len": 88,
          "_name": "btn_1w0oilu",
        },
      },
      "theme": Object {
        "color": Object {
          "primary": "#FF5A5F",
          "secondary": "#00A699",
        },
      },
    },
    "ref": null,
    "rendered": "hi",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "hi",
        "css": [Function],
        "onClick": [MockFunction],
        "styles": Object {
          "btn": Object {
            "_definition": Object {
              ":hover": Object {
                "backgroundColor": "orange",
              },
              "fontSize": "3em",
              "margin": 1,
              "padding": ".2em 1em",
            },
            "_len": 88,
            "_name": "btn_1w0oilu",
          },
        },
        "theme": Object {
          "color": Object {
            "primary": "#FF5A5F",
            "secondary": "#00A699",
          },
        },
      },
      "ref": null,
      "rendered": "hi",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
